osu.Game/Storyboards/Storyboard.cs

```diff
@@ -46,14 +50,17 @@ public bool ReplacesBackground
                 if (backgroundPath == null)
                     return false;
 
-                return GetLayer("Background").Elements.Any(e => e.Path.ToLowerInvariant() == backgroundPath);
+                if (GetLayer(LegacyStoryLayer.Video).Elements.Any())
+                    return true;
+
+                return GetLayer(LegacyStoryLayer.Background).Elements.Any(e => e.Path.ToLowerInvariant() == backgroundPath);
             }
         }
 
         public DrawableStoryboard CreateDrawable(WorkingBeatmap working = null)
         {
             var drawable = new DrawableStoryboard(this);
-            drawable.Width = drawable.Height * (BeatmapInfo.WidescreenStoryboard ? 16 / 9f : 4 / 3f);
+            drawable.Width = drawable.Height * (BeatmapInfo.WidescreenStoryboard || GetLayer(LegacyStoryLayer.Video).Elements.Any() ? 16 / 9f : 4 / 3f);
```

Comment:
where does this come from?

Reply:
This was added to prevent the storyboard from masking the video to 4:3

Reply:
It looks like masking isn't actually implemented right now (it's just limiting the size).

The correct solution here is to move the masking (sizing?) logic to `DrawableStoryboardLayer` and actually apply it (by setting `Masking = true`). Probably best done as a separate PR. Then the video layer could choose in its constructor to not be masked.

Reply:
`Masking = true` is set for the entire storyboard in [DimmableStoryboard](            https://github.com/ppy/osu/blob/8fb41e4c670f1dcf1b8b27d7168d89b67e8d6f43/osu.Game/Screens/Play/DimmableStoryboard.cs#L47)

With this removed, the video will always display at full brightness (ignoring `UserDimContainer`)
It appears `DrawableStoryboardVideo` will render with the dim as long as any of its parents have masking enabled (including `DimmableStoryboard` itself)

Should this part be left as-is for now?

Reply:
Aha. That masking can be removed after the above refactor is made.

Masking should not affect dim, though.

Reply:
It looks like `VideoSprite` must have masking in any of its parents in order for dims to apply. Are there any special properties about either masking or the `VideoSprite` which would cause this?
The existing `DimmableVideo` implementation also has `Masking = true` set even though nothing should actually get masked away.

Reply:
This shouldn't be the case and requires further investigation. We shouldn't be working around this.

Reply:
Is an appropriate solution to allow `UserDimContainer` to use a `BufferedContainer`?

Reply:
This is probably a framework issue which I initially caught but thought it had been fixed, relating to the non-masking shader of `VideoSprite`. Will look into it and fix.

Reply:
Please test again using the referencing o!f PR.

Reply:
Works great!
Should masking be reimplemented per-layer in a separate PR this one depends on?